src/store/lokapi.ts

```diff
@@ -116,7 +116,12 @@ export var moduleLokAPI = {
           }, '');
         }
         // accounts can be an empty array
-        state.accounts = accounts
+        for (var i = 0; i < accounts.length; i++) {
+          state.accounts[i] = { name: '', bal: 0, curr: '' }
+          state.accounts[i].name = accounts[i].internalId
+          state.accounts[i].bal = parseFloat(await accounts[i].getBalance())
+          state.accounts[i].curr = await accounts[i].getSymbol()
```

Comment:
shouldn't we try to avoid triggering watchers and proxies by doing 4 modification of the state were we could do only one ? and use `push` ? `state.accounts.push({...})`. Or you wanted to have a sort of a placeholder while it was resolving bar and curr ? Then at least you could set the name directly as `internalId` is not waiting for a promise.

Here are some example of things we should probably talk about. The `internalId` is named internal because it is mainly technical. I'm not sure we want to display that to the user, but I completely understand why you went this way as there is something missing here and I'm not sure what is the best way to deal with this. Technically, we have a kind of tree: Backend (ie: cyclos, comchain) --o2m--> Currency (ie: cyclos on this server, comchain on a currency) --o2m--> userAccount (my account on this server on cyclos, or my wallet in this currency on comchain) --o2m--> bankAccounts

But there are a lot of these notions that are slightly different in cyclos and comchain.

Anyway, we could give the backend the opportunity to try to give you a displayName for an account. But in both comchain or cyclos, this is not really offered... this might need to come from odoo, as it is already the case for naming destination accounts in comchain (and actually should work the same in cyclos).

Reply:
@vaab 
> Shouldn't we try to avoid triggering watchers and proxies by doing 4 modification of the state were we could do only one ? and use `push` ? `state.accounts.push({...})`

Yes, you're absolutely right.

> I don't see a valid key used to help vuejs identify changes, and you seem to rely on the fact that accounts from getAccounts in lokapi will be returned in the same order each time. Or you wanted to have a sort of a placeholder while it was resolving bar and curr ? Then at least you could set the name directly as internalId is not waiting for a promise.

Also all correct. I was being too quick. Thanks for the heads-up.